File: Introductory_Problems/Bit_Strings.cpp
#include <bits/stdc++.h>
using namespace std;

long long fastModularExpo(long long base, long long expo) {
    long long MOD = 1e9 + 7;
    int res = 1;
    while (expo > 0) {
        if (expo & 1LL) {
            res = (res*base) % MOD;
        }
        base = (base*base) % MOD;
        expo >>= 1LL;
    }
    return res;
}

int main() {
    int n ;
    cin >> n;
    
    cout << fastModularExpo(2, n);
    return 0;
}

File: Introductory_Problems/Bit_strings.py
#Fast Modular Exponentiation

def bit_strings(expo):
    MOD = int(1e9)+7
    base = 2
    result = 1
    while (expo > 0) :
        if (expo & 1) :
            result = (result * base)%MOD 
        base = (base*base)%MOD
        expo >>= 1
    return result


if __name__ == "__main__" :
    n = int(input())
    print(bit_strings(n))
    
# Mistake : Used 1e9 + 7 instead of int(1e9) + 7
# And Use mod on the base * base operation